src/app/component/eco-news/components/create-edit-news/create-edit-news.component.ts

```diff
@@ -232,11 +232,13 @@ export class CreateEditNewsComponent extends FormBaseComponent implements OnInit
   public setActiveFilters(itemToUpdate: EcoNewsModel): void {
     if (itemToUpdate.tags.length) {
       this.isArrayEmpty = false;
-
       itemToUpdate.tags.forEach((tag: NewsTagInterface) => {
-        const index = this.filters.findIndex((filterObj: FilterModel) => filterObj.name === tag.name);
-
-        this.filters = [...this.filters.slice(0, index), { name: tag.name, isActive: true }, ...this.filters.slice(index + 1)];
+        const index = this.filters.findIndex((filterObj: FilterModel) => filterObj.name === `${tag}`);
+        this.filters = [
+          ...this.filters.slice(0, index),
+          { name: `${tag}`, isActive: true },
+          ...this.filters.slice(index + 1)
+        ];
```

Comment:
you have the same logic on lines 282-286. move it to another method

Reply:
the repeating logic has been refactored with this commit 06b50aa07c28f4efdd5bf5a7b88b0a038a213903